spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -269,13 +278,24 @@ public void restartJobExecution(long jobExecutionId) throws NoSuchJobExecutionEx
 	 *
 	 * @param taskExecutionArgs original set of task execution arguments
 	 * @param jobParameters     for the job to be restarted.
+	 * @param useJsonJobParameters determine what converter to use to serialize the job parameter to the command line arguments.
 	 * @return deduped list of arguments that contains the original arguments and any
 	 * identifying job parameters not in the original task execution arguments.
 	 */
-	private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobParameters jobParameters) {
+	private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobParameters jobParameters,
+		Boolean useJsonJobParameters) {
```

Comment:
This method can be simplified a bit via stream api - something like this:
```java
		if (useJsonJobParameters == null) {
			useJsonJobParameters = taskConfigurationProperties.isUseJsonJobParameters();
		}
		var jobParamsConverter = useJsonJobParameters ? new ScdfJsonJobParametersConverter()
				: new ScdfDefaultJobParametersConverter();
		List<String> result = new ArrayList<>(taskExecutionArgs);
		jobParameters.getParameters().entrySet().stream()
				.filter((e) -> !e.getKey().startsWith("-"))
				.filter((e) -> !taskExecutionArgs.stream().anyMatch((arg) -> arg.startsWith(e.getKey())))
				.map((e) -> e.getKey() + "=" + jobParamsConverter.deserializeJobParameter(e.getValue()))
				.forEach(result::add);
		return result;
```

Wdyt?

Reply:
LGTM.   Tweaked it to use `noneMatch` instead of `!taskExecutionArgs.stream().anyMatch` 
Will merge change when review is complete.